app/services/checksum_validator.rb

```diff
@@ -1,6 +1,15 @@
 # code for validating Moab checksums
 class ChecksumValidator
   include ::MoabValidationHandler
+  include ActiveModel::Validations
```

Comment:
It seemed preferable to add in validations and perform it in the initialize method, since there are multiple public methods that could be called ...